apps/ledger-live-desktop/src/newArch/WalletSync/Flows/Activation/03-ActivationOrSynchroWithTrustchain.tsx

```diff
@@ -1,43 +1,29 @@
-import React, { useEffect } from "react";
-import { useDispatch } from "react-redux";
+import React from "react";
 import { Device } from "@ledgerhq/live-common/hw/actions/types";
-import { Flow, Step } from "~/renderer/reducers/walletSync";
-import { setFlow } from "~/renderer/actions/walletSync";
 
 import FollowStepsOnDevice from "../DeviceActions/FollowStepsOnDevice";
 import ErrorDisplay from "~/renderer/components/ErrorDisplay";
 import { useAddMember } from "../../hooks/useAddMember";
+import { InfiniteLoader } from "@ledgerhq/react-ui";
 
 type Props = {
   device: Device | null;
 };
 
 export default function ActivationOrSynchroWithTrustchain({ device }: Props) {
-  const dispatch = useDispatch();
+  const { error, userDeviceInteraction, handleMissingDevice, onRetry } = useAddMember({ device });
 
-  const onRetry = () => {
-    dispatch(setFlow({ flow: Flow.Activation, step: Step.DeviceAction }));
-  };
-
-  const addMemberMutation = useAddMember({ device });
-
-  useEffect(() => {
-    addMemberMutation.mutate();
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, []);
+  if (!device) {
+    handleMissingDevice();
+  }
 
-  if (device) {
-    return addMemberMutation.isError ? (
-      <ErrorDisplay error={addMemberMutation.error} withExportLogs onRetry={onRetry} />
+  if (userDeviceInteraction && device) {
+    return error ? (
```

Comment:
to me, error can occur regardless of `userDeviceInteraction` so it's maybe better to have a first check `if (error) return <ErrorDisplay` before the other ifs